app/models/gym_membership.rb

```diff
@@ -3,7 +3,14 @@ class GymMembership < ApplicationRecord
 
   has_many :events, dependent: :destroy
 
+  delegate :address, to: :find_gym
+  delegate :address_details, to: :find_gym
```

Comment:
I'm not sure this is the best use of delegation here because `find_gym` isn't a model.  But it is a creative solution!  We'll need to have a think on how to reorganize/de-hackify this in later development.